convert2rhel/checks.py

```diff
@@ -422,10 +422,19 @@ def _bad_kernel_version(kernel_release):
 
 def _bad_kernel_package_signature(kernel_release):
     """Return True if the booted kernel is not signed by the original OS vendor, i.e. it's a custom kernel."""
-    kernel_pkg = run_subprocess(
-        ["rpm", "-qf", "--qf", "%{NAME}", "/boot/vmlinuz-%s" % kernel_release], print_output=False
-    )[0]
+    vmlinuz_path = "/boot/vmlinuz-%s" % kernel_release
+    kernel_pkg, return_code = run_subprocess(["rpm", "-qf", "--qf", "%{NAME}", vmlinuz_path], print_output=False)
     logger.debug("Booted kernel package name: {0}".format(kernel_pkg))
+
+    if return_code == 1:
+        os_vendor = system_info.name.split()[0]
```

Comment:
You have the same line that gets the os_vendor on line 442, I believe that you can remove that call from line 442 and move your `os_vendor` variable to be above that if statement `if return_code == 1`, like this: 

```suggestion
        # e.g. Oracle Linux Server -> Oracle
        os_vendor = system_info.name.split()[0]
        if return_code == 1:
```